src/server/api/routes/users.js

```diff
@@ -46,6 +55,25 @@ router.get('/users/self', async (req, res, next) => {
   }
 });
 
+router.post('/users/self/update', async (req, res, next) => {
+  try {
+    const { user } = req.session;
+    if (user) {
+      const { hasReadFirstTimeSignInDialog } = req.body;
+      await userCollection.doc(user).update({
+        hasReadFirstTimeSignInDialog,
+      });
+      res.sendStatus(200);
+      return;
+    }
+    res.sendStatus(404);
+  } catch (err) {
+    if (req.session) req.session = null;
+    res.clearCookie(AUTH_COOKIE_NAME, CLEAR_AUTH_COOKIE_OPTION);
```

Comment:
`/users/self` was decided to be called every time user reach our site, so it also handle invalid session
for other (authed) API endpoint, just throw 403 or slient ignore if session is not right, no need to handle